pretix_eth/exporter.py

```diff
@@ -56,57 +127,16 @@ def iterate_list(self, form_data):
 
         objs = sorted(list(payments) + list(refunds), key=lambda obj: obj.created)
 
-        headers = [
-            'Type', 'Event slug', 'Order', 'Payment ID', 'Creation date',
-            'Completion date', 'Status', 'Amount', 'Token', 'Wallet address'
-        ]
-
-        yield headers
+        yield self.headers
 
         yield self.ProgressSetTotal(total=len(objs))
         for obj in objs:
-            time_zone = pytz.timezone(obj.order.event.settings.timezone)
-            if isinstance(obj, OrderPayment) and obj.payment_date:
-                completion_date = obj.payment_date.astimezone(
-                    time_zone).date().strftime('%Y-%m-%d')
-            elif isinstance(obj, OrderRefund) and obj.execution_date:
-                completion_date = obj.execution_date.astimezone(
-                    time_zone).date().strftime('%Y-%m-%d')
-            else:
-                completion_date = ''
-
             if isinstance(obj, OrderPayment):
-                token = obj.info_data['currency_type']
+                row = self.payment_to_row(obj)
             elif isinstance(obj, OrderRefund):
-                token = obj.payment.info_data['currency_type']
+                row = self.refund_to_row(obj)
             else:
-                token = ''
-
-            if isinstance(obj, OrderPayment) and WalletAddress.objects.filter(
-                    order_payment=obj).first():
-                wallet_address = WalletAddress.objects.filter(
-                    order_payment=obj).first().hex_address
-            elif isinstance(obj, OrderRefund) and WalletAddress.objects.filter(
-                    order_payment=obj.payment).first():
-                wallet_address = WalletAddress.objects.filter(
-                    order_payment=obj.payment).first().hex_address
-            else:
-                wallet_address = ''
-
-            object_type = 'Payment' if isinstance(obj, OrderPayment) else 'Refund'
-
-            row = [
-                object_type,
-                obj.order.event.slug,
-                obj.order.code,
-                obj.full_id,
-                obj.created.astimezone(time_zone).date().strftime('%Y-%m-%d'),
-                completion_date,
-                obj.state,
-                obj.amount,
-                token,
-                wallet_address
-            ]
+                raise Exception('Invariant: unexpected type in ethereum data exporter')
```

Comment:
You might as well include the type in the error string here so that if you run into this exception in server logs you can know what unexpected object got passed in.